services/brig/src/Brig/API/Client.hs

```diff
@@ -131,6 +137,11 @@ updateClient u c r = do
   unless ok $
     throwE ClientNotFound
   for_ (updateClientLabel r) $ lift . Data.updateClientLabel u c . Just
+  for_ (updateClientSupportedFeatures r) $ \features' -> do
+    features <- fromMaybe [] <$> Data.lookupClientSupportedFeatures u c
+    if Set.fromList features `Set.isSubsetOf` Set.fromList features'
+      then lift . Data.updateClientSupportedFeatures u c . Just $ features'
+      else throwE ClientFeaturesCannotBeRemoved
```

Comment:
This doesn't make sense. Of course features can be removed, or changed, e.g. support for implicit consent could be removed, and instead support for explicit consent added. Why force clients in 3 years to support half-implemented features from the distant past?

Reply:
It does make sense if we see this as a data-minimizing substitute for version information.  Versions usually aren't decreasing either, and I think it's reasonable to disallow it.

Reply:
Another thing that can be said about in favor of disallowing it is that the restriction can be trivially lifted whenever we change our mind.

Reply:
Okay, that's true, this check can be removed again.